src/main/java/uk/gov/pay/directdebit/webhook/gocardless/support/GoCardlessWebhookVerifier.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.pay.directdebit.webhook.gocardless.support;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.pay.directdebit.webhook.gocardless.exception.InvalidWebhookException;
+
+import static java.lang.String.format;
+
+public class GoCardlessWebhookVerifier {
+    
+    private final GoCardlessWebhookSignatureCalculator goCardlessWebhookSignatureCalculator;
+
+    public GoCardlessWebhookVerifier(String secret) {
+        goCardlessWebhookSignatureCalculator = new GoCardlessWebhookSignatureCalculator(secret);
```

Comment:
This doesn't really follow dependency injection with the `new` in the constructor. We can `@Inject` it in and create the binding within `DirectDebitModule` using: 
```
@Provides
    @Singleton
    public GoCardlessWebhookSignatureCalculator provideGoCardlessWebhookSignatureCalculator() {
        if (StringUtils.isBlank(configuration.getGoCardless().getWebhookSecret())) {
            throw new RuntimeException("GoCardless webhook secret is blank");
        }
        return new GoCardlessWebhookSignatureCalculator(configuration.getGoCardless().getWebhookSecret());
    }
```
To check it works before suggesting it I made the changes, if you want to see them I could add a commit or put onto a throw-away branch for your perusal. As part of the change this method can go from `GoCardlessFactory`

```
public GoCardlessWebhookVerifier buildSignatureVerifier() {
        if (StringUtils.isBlank(webhookSecret)) {
            throw new RuntimeException("GoCardless webhook secret is blank");
        }

        return new GoCardlessWebhookVerifier(webhookSecret);
    }
```

Reply:
Made this change.